jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -159,47 +219,114 @@ def _on_spectral_unit_changed(self, *args):
                                self.spectral_unit.selected,
                                sender=self))
 
-    @observe('flux_unit_selected')
+    @observe('flux_unit_selected', 'sb_unit_selected')
     def _on_flux_unit_changed(self, *args):
+        flux_or_sb = None
+        current_y = self.spectrum_viewer.state.y_display_unit
 
-        yunit = _valid_glue_display_unit(self.flux_or_sb_unit.selected, self.spectrum_viewer, 'y')
+        data_collection_unit = ''
+        # need to determine the input spectrum units to disable the additional
+        # drop down and possiblity of translations
+        if self.app.data_collection[0] and self.app.config == 'specviz':
+            if check_if_unit_is_per_solid_angle(self.app.data_collection[0].get_object()._unit):
+                data_collection_unit = 'Surface Brightness'
+                self.specviz_disabler = 'Flux'
+            else:
+                self.specviz_disabler = 'Surface Brightness'
+                data_collection_unit = 'Flux'
 
-        if self.spectrum_viewer.state.y_display_unit != yunit:
+        for arg in args:
+            # determine if flux or surface brightness unit was changed  by user
+            if arg['name'] == 'flux_unit_selected':
+                if data_collection_unit == 'Surface Brightness':
+                    raise ValueError(
+                        "Unit translation between Flux and Surface Brightness "
+                        "is not supported in Specviz."
+                    )
+                flux_or_sb = self.flux_unit.selected
+                # update flux or surface brightness dropdown if necessary
+                if check_if_unit_is_per_solid_angle(current_y):
+                    self.flux_or_sb.selected = 'Flux'
+
+                untranslatable_units = self.untranslatable_units()
+                # disable translator is flux unit is untranslatable
+                # still can select surface brightness units, just disables
+                # flux -> surface brightnes of particular unit selected.
+                if flux_or_sb in untranslatable_units:
+                    self.can_translate = False
+                else:
+                    self.can_translate = True
 
+            elif arg['name'] == 'sb_unit_selected':
+                if data_collection_unit == 'Flux':
+                    raise ValueError(
+                        "Unit translation between Flux and Surface Brightness "
+                        "is not supported in Specviz."
+                    )
```

Comment:
I believe so, I do update the API so that the traitlets to the other physical type aren't exposed, these raise statements are there in case someone tries to translate with `._obj`.

Reply:
I'm not sure this will be relevant anymore if getting rid of `specviz_disabler`, but if it is (or we can't get rid of `specviz_disabler`), then maybe just using `{self.app.config}` would be safer just in case this can get triggered in another config?